Code/GraphMol/SmilesParse/SmilesParseOps.cpp

```diff
@@ -447,8 +432,8 @@ void CloseMolRings(RWMol *mol, bool toleratePartials) {
           Bond *bond2 = *bondIt;
 
           // remove those bonds from the bookmarks:
-          mol->clearBondBookmark(bookmarkIt->first, bond1);
-          mol->clearBondBookmark(bookmarkIt->first, bond2);
+          mol->clearBondBookmark(bookmark.first, bond1);
```

Comment:
Does this invalidate the iterator?

Reply:
same story here, the loop is over atom bookmarks